src/chainstate/burn/db/sortdb.rs

```diff
@@ -3801,6 +3992,20 @@ impl<'a> SortitionHandleTx<'a> {
         let apparent_sender_str = serde_json::to_string(&block_commit.apparent_sender)
             .map_err(|e| db_error::SerializationError(e))?;
 
+        // find parent block commit's snapshot's sortition ID.
+        // If the parent_block_ptr doesn't point to a valid snapshot, then store an empty
+        // sortition.  If we're not testing, then this should never happen.
```

Comment:
This should never happen because the `parent_block_ptr` is checked for validity elsewhere? Can you expand this comment to specify where that check occurs?

Reply:
I updated the comment.  The reason is that `insert_block_commit()` gets called only for valid block-commits.  One of the validity checks is that a block-commit's parent must have been accepted and stored already in this sortition history, or it must point to `(0,0)`.  This check happens in `LeaderBlockCommitOp::check()`.